android/src/main/java/ly/count/android/sdk/react/CountlyReactNative.java

```diff
@@ -1680,6 +1680,30 @@ public Map<String, Object> convertToEventMap(ReadableArray segments) {
                             segmentation.put(key, doubleValue);
                         }
                         break;
+                    case Array:
+                        ReadableArray arrayValue = segments.getArray(i + 1);
```

Comment:
isnt underlying sdk will do this anyway?

Reply:
I could not find a way where I was able to remove this check and made sure we still support arrays